atc/scheduler/factory/build_factory_test.go

```diff
@@ -0,0 +1,679 @@
+package factory_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/atc/db"
+	"github.com/concourse/concourse/atc/scheduler/factory"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"sigs.k8s.io/yaml"
+)
+
+type BuildFactorySuite struct {
+	suite.Suite
+	*require.Assertions
+}
+
+type FactoryTest struct {
+	Title string
+
+	ConfigYAML string
+	Inputs     []db.BuildInput
+
+	CompareIDs bool
+	PlanJSON   string
+}
+
+var resources = atc.ResourceConfigs{
+	atc.ResourceConfig{
+		Name:   "some-resource",
+		Type:   "some-resource-type",
+		Source: atc.Source{"some": "source"},
+	},
+}
+
+var resourceTypes = atc.VersionedResourceTypes{
+	{
+		ResourceType: atc.ResourceType{
+			Name:   "some-resource-type",
+			Type:   "some-base-resource-type",
+			Source: atc.Source{"some": "type-source"},
+		},
+		Version: atc.Version{"some": "type-version"},
+	},
+}
+
+var factoryTests = []FactoryTest{
+	{
+		Title: "get step",
+		ConfigYAML: `
+			get: some-name
+			resource: some-resource
+			params: {some: params}
+			version: {doesnt: matter}
```

Comment:
Just to confirm my understanding: this version doesn't matter, since it's the responsibility of the scheduler (algorithm?) to determine the build inputs - so this pinned version in the config would be used by the scheduler, and the `BuildFactory` takes in whatever `BuildInputs` are given to it as the source of truth?

Reply:
Yep!